### ISSUE TITLE
[analysis] wrong `:name-col` and `:name-end-col` for `declare` var-definitions

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1342]
**version**

2021-07-28

**platform**

JVM

**editor**

tested with clojure-lsp and clj-kondo from CLI

**problem**

clj-kondo return the analysis of var-definition for `declare` with wrong `:name-col`, it returns the `:name-col` as the same of `:col` making impossible to know where the symbol name starts.
This cause multiple issues on clojure-lsp like document highlighting and rename not renaming correctly: https://github.com/clojure-lsp/clojure-lsp/issues/497

**repro**

```bash
echo '(declare foo)' | clj -Sdeps '{:deps {clj-kondo/clj-kondo {:mvn/version "2021.07.28"}}}' -m clj-kondo.main  --lint "-" --config '{:cache true :copy-configs true :output {:analysis {:keywords true} :format :json :canonical-paths true}}' |jq .analysis
```

```json
...
{
      "end-row": 1,
      "name-end-col": 14,
      "name-end-row": 1,
      "name-row": 1,
      "ns": "user",
      "name": "foo",
      "defined-by": "clojure.core/declare",
      "filename": "<stdin>",
      "col": 1,
      "name-col": 1,
      "end-col": 14,
      "row": 1
    }
...
```

**expected behavior**

Follow other analysis returning the correct `:name-col` and `:name-end-col` of the other symbols.

```json
...
{
      "end-row": 1,
      "name-end-col": 13,
      "name-end-row": 1,
      "name-row": 1,
      "ns": "user",
      "name": "foo",
      "defined-by": "clojure.core/declare",
      "filename": "<stdin>",
      "col": 1,
      "name-col": 10,
      "end-col": 14,
      "row": 1
    }
...
```
